internal/grafanadi/handler/podsummary_handler.go

```diff
@@ -0,0 +1,205 @@
+package handler
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/alipay/container-observability-service/internal/grafanadi/model"
+	"github.com/alipay/container-observability-service/pkg/dal/storage-client/data_access"
+	storagemodel "github.com/alipay/container-observability-service/pkg/dal/storage-client/model"
+
+	"github.com/alipay/container-observability-service/pkg/slo"
+	"github.com/alipay/container-observability-service/pkg/utils"
+	interutils "github.com/alipay/container-observability-service/internal/grafanadi/utils"
+	"github.com/alipay/container-observability-service/pkg/metrics"
+	"github.com/alipay/container-observability-service/internal/grafanadi/service"
+)
+
+type PodSummaryHandler struct {
+	request       *http.Request
+	writer        http.ResponseWriter
+	requestParams *PodSummaryParams
+	storage       data_access.StorageInterface
+}
+
+type PodSummaryParams struct {
+	PodUIDName string
+	PodUID     string
+}
+
+func (handler *PodSummaryHandler) RequestParams() interface{} {
+
+	return handler.requestParams
+}
+
+func (handler *PodSummaryHandler) ParseRequest() error {
+	params := PodSummaryParams{}
+	if handler.request.Method == http.MethodGet {
+		params.PodUIDName = handler.request.URL.Query().Get("searchkey")
+		params.PodUID = handler.request.URL.Query().Get("searchvalue")
+	}
+
+	handler.requestParams = &params
+	return nil
+}
+
+func (handler *PodSummaryHandler) ValidRequest() error {
+	return nil
+}
+
+func (handler *PodSummaryHandler) QueryPodSummary(key, value string) (int, interface{}, error) {
+	podInfos := make([]*storagemodel.PodInfo, 0)
+	var podSummary = model.PodSummary{}
+	slotracedata := make([]*storagemodel.SloTraceData, 0)
+	lifephases := make([]*storagemodel.LifePhase, 0)
+	podYamls := make([]*storagemodel.PodYaml, 0)
+	if value == "" {
+		return http.StatusOK, nil, nil
+	}
+
+	begin := time.Now()
+	defer func() {
+		cost := utils.TimeSinceInMilliSeconds(begin)
+		metrics.QueryMethodDurationMilliSeconds.WithLabelValues("QueryPodSummary").Observe(cost)
+	}()
+	util := interutils.Util{
+		Storage: handler.storage,
+	}
+	util.GetUid(podYamls, key, &value)
+	podUid := value
+
+	var err error
+	var wg sync.WaitGroup
+	wg.Add(3)
+	go func() {
+		defer wg.Done()
+		err = handler.storage.QueryPodInfoWithPodUid(&podInfos, podUid)
+	}()
+	go func() {
+		defer wg.Done()
+		err = handler.storage.QuerySloTraceDataWithPodUID(&slotracedata, podUid)
+	}()
+	go func() {
+		defer wg.Done()
+		err = handler.storage.QueryLifePhaseWithPodUid(&lifephases, podUid)
+	}()
+	if err != nil {
```

Comment:
```suggestion
	if err != nil {
```

Reply:
Assigning incorrect values to multiple go functions can indeed lead to inconsistent results. Therefore, I modified the code to set separate error variables for each go func, and then judged the errors after all go funcs were completed.
```golang
        go func() {
		defer wg.Done()
		podInfoErr = handler.storage.QueryPodInfoWithPodUid(&podInfos, podUid)
	}()
	go func() {
		defer wg.Done()
		sloTraceDataErr = handler.storage.QuerySloTraceDataWithPodUID(&slotracedata, podUid)
	}()
	go func() {
		defer wg.Done()
		lifePhaseErr = handler.storage.QueryLifePhaseWithPodUid(&lifephases, podUid)
	}()
        wg.Wait()
	if podInfoErr != nil {
		return http.StatusBadRequest, nil, errors.New(podInfoErr.Error())
	}
	if sloTraceDataErr != nil {
		return http.StatusBadRequest, nil, errors.New(sloTraceDataErr.Error())
	}
	if lifePhaseErr != nil {
		return http.StatusBadRequest, nil, errors.New(lifePhaseErr.Error())
	}
```